pkg/auth/kv_migrate_test.go

```diff
@@ -0,0 +1,271 @@
+package auth_test
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"math/rand"
+	"testing"
+
+	"github.com/rs/xid"
+	"github.com/stretchr/testify/require"
+	"github.com/treeverse/lakefs/pkg/auth"
+	"github.com/treeverse/lakefs/pkg/auth/crypt"
+	"github.com/treeverse/lakefs/pkg/auth/model"
+	authparams "github.com/treeverse/lakefs/pkg/auth/params"
+	"github.com/treeverse/lakefs/pkg/kv"
+	"github.com/treeverse/lakefs/pkg/kv/kvtest"
+	"github.com/treeverse/lakefs/pkg/kv/postgres"
+	"github.com/treeverse/lakefs/pkg/logging"
+	"github.com/treeverse/lakefs/pkg/testutil"
+)
+
+func TestMigrate(t *testing.T) {
+	ctx := context.Background()
+	database, _ := testutil.GetDB(t, databaseURI)
+	kvStore := kvtest.MakeStoreByName(postgres.DriverName, databaseURI)(t, ctx)
+	defer kvStore.Close()
+	dbAuthService := auth.NewDBAuthService(database, crypt.NewSecretStore([]byte("someSecret")), authparams.ServiceCache{
+		Enabled: false,
+	}, logging.Default())
+
+	prepareTestData(t, ctx, dbAuthService)
+
+	buf := bytes.Buffer{}
+	require.NoError(t, auth.Migrate(ctx, database.Pool(), &buf))
+
+	testutil.MustDo(t, "Import migrated auth data", kv.Import(ctx, &buf, kvStore))
+	kvAuthService := auth.NewKVAuthService(kv.StoreMessage{Store: kvStore}, crypt.NewSecretStore([]byte("someSecret")), authparams.ServiceCache{
+		Enabled: false,
+	}, logging.Default())
+
+	verifyMigrationResults(t, ctx, dbAuthService, kvAuthService)
+}
+
+const MaxCredsPerUser = 4
```

Comment:
consolidate const declarations and move to the top of the module

Reply:
Done